py/modmath.c

```diff
@@ -132,7 +132,24 @@ MATH_FUN_1(asin, asin)
 // atan(x)
 MATH_FUN_1(atan, atan)
 // atan2(y, x)
+#if defined(_MSC_VER) && (_MSC_VER <= 1800)
+// VC++ 12.0 and earlier return NaN if one of the arguments is Inf.
+mp_float_t msvc_atan2(mp_float_t x, mp_float_t y) {
+    if (isinf(x) && isinf(y)) {
+        static const mp_float_t pi_over_four = MICROPY_FLOAT_CONST(M_PI_4);
+        static const mp_float_t three_pi_over_four = MICROPY_FLOAT_CONST(3.0) * MICROPY_FLOAT_CONST(M_PI_4);
```

Comment:
Will this introduce some error in computing `3pi/4` due to the mult?  And will it do the mult at runtime?  Might be better to actually write the full constant here...?

Reply:
The constants are calculated at compile time for msvc, no idea if standard guarantees that though so to make sure a constant would be better. No experience with that in practice, so the question is: what is an authorative source for that constant? Also to answer the question wrt whether the compile-time calculated value has an error; the constant I get now is 2.3561944901923448. Which also happens to have a lot of search engine matches but that doesn't mean much.

Reply:
The value to lots of decimal places is `2.356194490192344928846982537459627163147877049531329365731`.  I tested computing it via `3.0 * M_PI / 4.0` and `3.0 * M_PI_4` (actual computation at runtime) and it gets the right answer.  But I wouldn't want to rely on this.

It looks like it's enough to truncate it at `2.3561944901923449` but it might be good to use `2.35619449019234492885` to match the precision of `MP_PI` defined at the top of `modmath.c`.

Reply:
Ok but I'm interested in knowing how you get a constant like that. Plus I need the M_PI_4 one itself as well.

Reply:
You can try https://www.wolframalpha.com/ .  Or find PI somewhere on the internet to many places (eg the benchmark in this repo called bm_pidigts.py), convert to an integer (a very large one like 31415926), then apply `//4` or whatever factor is needed as an integer operation.

Reply:
Ha, forgot about wolframalpha. Great site.